src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import core.basesyntax.service.FruitConverter;
+import core.basesyntax.service.exceptions.InvalidDataException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int NUMBER_OF_COLUMNS = 3;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> inputReport) {
+        return inputReport.stream()
+                .map(a -> a.trim())
+                .map(a -> a.split(","))
+                .filter(a -> isDataCorrect(a))
+                .map(el -> {
+                    if (isInteger(el[2].trim())) {
+                        FruitTransaction transaction = new FruitTransaction();
+                        transaction.setOperation(FruitTransaction.Operation.fromCode((el[0])));
+                        transaction.setFruit(FruitConverter.convertToFruit(el[1]));
+                        transaction.setQuantity(Integer.valueOf(el[2]));
+                        return transaction;
+                    } else {
+                        return null;

```

Comment:
Returning null in the stream can lead to NullPointerException in the future. It would be better to throw an exception in case of invalid data.

Reply:
ok, done